src/qtfirebasedb.h

```diff
@@ -0,0 +1,199 @@
+#ifndef QTFIREBASEDATABASE_H
+#define QTFIREBASEDATABASE_H
+
+#include "qtfirebaseservice.h"
+#include "firebase/database.h"
+#include <QMutex>
+
+#ifdef QTFIREBASE_BUILD_DATABASE
+#include "src/qtfirebase.h"
+#if defined(qFirebaseDb)
+#undef qFirebaseDb
+#endif
+#define qFirebaseDb (static_cast<QtFirebaseDb*>(QtFirebaseDb::instance()))
+
+class QtFirebaseDbRequest;
+class QtFirebaseDb : public QtFirebaseService
+{
+    Q_OBJECT
+    typedef QSharedPointer<QtFirebaseDb> Ptr;
+public:
+    static QtFirebaseDb* instance() {
+        if(self == 0) {
+            self = new QtFirebaseDb(0);
+            qDebug() << self << "::QtFirebaseDb" << "singleton";
+        }
+        return self;
+    }
+
+    enum Error
+    {
+        kErrorNone = firebase::database::kErrorNone,
+        kErrorDisconnected = firebase::database::kErrorDisconnected,
+        kErrorExpiredToken = firebase::database::kErrorExpiredToken,
+        kErrorInvalidToken = firebase::database::kErrorInvalidToken,
+        kErrorMaxRetries = firebase::database::kErrorMaxRetries,
+        kErrorNetworkError = firebase::database::kErrorNetworkError,
+        kErrorOperationFailed = firebase::database::kErrorOperationFailed,
+        kErrorOverriddenBySet = firebase::database::kErrorOverriddenBySet,
+        kErrorPermissionDenied = firebase::database::kErrorPermissionDenied,
+        kErrorUnavailable = firebase::database::kErrorUnavailable,
+        kErrorUnknownError = firebase::database::kErrorUnknownError,
+        kErrorWriteCanceled = firebase::database::kErrorWriteCanceled,
+        kErrorInvalidVariantType = firebase::database::kErrorInvalidVariantType,
+        kErrorConflictingOperationInProgress = firebase::database::kErrorConflictingOperationInProgress,
+        kErrorTransactionAbortedByUser = firebase::database::kErrorTransactionAbortedByUser,
+    };
+    Q_ENUM(Error)
+
+    //TODO implement database related functions
```

Comment:
What are these? Can you provide a link here maybe?

Reply:
https://firebase.google.com/docs/reference/cpp/class/firebase/database/database